File: ui/real-estate-ui/src/shared/types/property.ts
// src/shared/types/property.ts

export interface PropertyListDto {
  id: number;
  title: string;
  propertyType: string;
  propertyStatus: string;
  price: number;
  currency: string;
  thumbnailUrl: string;
  startDate: string;
  endDate?: string;
  adress?:string
}

export interface PropertyFilterDto {
  propertyTypeId?: number;
  propertyStatusId?: number;
  minPrice?: number;
  maxPrice?: number;
  startDate?: string;
  endDate?: string | null;
  city?: string | null;
  district?: string | null;
  query?: string | null;
  page: number;
  pageSize: number;
}



export interface PropertyFilterResponse {
  totalCount: number;
  page: number;
  pageSize: number;
  data: PropertyListDto[];
}

export interface PropertyTableRow {
  id: number;                // Table’da string olarak tutmak daha rahat, mapping’de çeviriyoruz
  title: string;
  propertyType: string;
  propertyStatus: string;
  price: number;
  currency: string;
  startDate: string;
  endDate?: string;
  thumbnailUrl: string;
}

export interface PropertyCreateDto {
  title: string;
  description: string;
  price: number;
  currencyId: number;
  propertyTypeId: number;
  propertyStatusId: number;
  startDate: string;
  endDate?: string;
  address?: string;
  latitude?: number;
  longitude?: number;
  photoUrls?: string[];
}

// (İsteğe bağlı) Güncelleme işlemleri için
export interface PropertyUpdateDto extends PropertyCreateDto {
  id: number;
}

export interface PropertyDetailDto {
  id: number;
  title: string;
  description: string;
  price: number;
  propertyType: string;
  propertyStatus: string;
  currency: string;
  address?: string;
  startDate: string;
  endDate?: string;
  thumbnailUrl?: string;
  photoUrls?: string[];

  m2Brut?: number;
  m2Net?: number;
  roomCount?: string;
  floor?: string;
  buildingAge?: number;

  ownerName?: string;
  ownerPhone?: string;
  latitude?: number;
  longitude?: number;
}

File: backend/RealEstateApp/RealEstateApp/RealEstateApp.Core/Entities/Property.cs
namespace RealEstateApp.Core.Entities
{
    public class Property
    {
        public int Id { get; set; }
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public decimal Price { get; set; }
        public int CurrencyId { get; set; }
        public Currency Currency { get; set; } = null!;
        public int PropertyTypeId { get; set; }
        public PropertyType PropertyType { get; set; } = null!;
        public int PropertyStatusId { get; set; }
        public PropertyStatus PropertyStatus { get; set; } = null!;
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string? Address { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public string? ThumbnailUrl { get; set; }
        public ICollection<PropertyPhoto>? Photos { get; set; }
        public int? M2Brut { get; set; }        
        public int? M2Net { get; set; }         
        public string? RoomCount { get; set; }  
        public int? BuildingAge { get; set; }
        public string? Floor { get; set; }

    }
}
